src/test/regress/expected/clock.out

```diff
@@ -252,8 +252,12 @@ DETAIL:  Key (cc)=((100,100)) already exists.
 --
 SELECT (extract(epoch from now()) * 1000)::bigint AS epoch,
 	citus_get_node_clock() AS latest_clock \gset
+-- Extract epoch from now() returns the epoch in seconds, where as citus_get_node_clock()
+-- takes epoch with microseconds precison and calculates milliseconds using the formula
+-- (seconds * 1000 + microseconds/1000), so we might lose milliseconds from the former,
+-- account for the worst case when checking.
 -- Returns true
-SELECT ABS(:epoch - cluster_clock_logical(:'latest_clock')) < 25;
+SELECT ABS(:epoch - cluster_clock_logical(:'latest_clock')) < 1000;
```

Comment:
I think the explanation makes sense. But having these type of queries return true/false makes them hard to debug. Can you change the query to return the actual value if it's higher than expected. An example of how to do this can be found here: https://github.com/citusdata/citus/blob/main/src/test/regress/flaky_tests.md#disk-size-numbers-are-not-exactly-like-expected

Reply:
Actually, the explanation seems to not match reality, `epoch from now()` has microsecond precision
```
> select extract(epoch from now());
      extract
───────────────────
 1670319132.353965
(1 row)
```
And the query above does:
```
> SELECT (extract(epoch from now()) * 1000)::bigint AS epoch,
        citus_get_node_clock() AS latest_clock;
     epoch     │   latest_clock
───────────────┼───────────────────
 1670319087583 │ (1670319087583,0)
(1 row)
```

Reply:
>>I noticed that clock test is being run in parallel with ~9 other tests. Could those concurrent tests impact this test somehow?

**SELECT (extract(epoch from now()) * 1000)::bigint AS epoch** << if the CPU is relinquished and re-scheduled after 25 ms  to complete this part >>  **citus_get_node_clock() AS latest_clock;** the test will fail.

Reply:
> if the CPU is relinquished and re-scheduled after 25 ms to complete this part >> citus_get_node_clock() ; the test will fail.

Makes sense, probably good to give to change the schedule to not run it in parallel with other tests then, to try and avoid re-scheduling. If that alone doesn't solve the problem (which seems likely given CI can be slow for no reason) I'd say it's fine to increase the allowed range some more.

Reply:
https://app.circleci.com/pipelines/github/citusdata/citus/29277/workflows/ae0127f2-d6d3-42ce-b789-05e7ec5e4d9e
https://app.circleci.com/pipelines/github/citusdata/citus/29277/workflows/e460527b-f7eb-422a-960a-fbd22b3c43be/jobs/911803

flaky debugging is passing, I suspect it has to do with the parallel schedule. Let's make it serial and see how it goes.